packages/headless/src/controllers/generated-answer/headless-generated-answer.test.ts

```diff
@@ -275,6 +275,26 @@ describe('generated answer', () => {
 
       expect(action).toBeTruthy();
     });
+
+    describe('when we have multiple controllers', () => {
+      let secondEngine: MockSearchEngine;
+      beforeEach(() => {
+        secondEngine = buildMockSearchAppEngine();
+        buildGeneratedAnswer(secondEngine);
+      });
+      it('should dispatch the stream action only once', () => {
+        engine.state.search.extendedResults = {
+          generativeQuestionAnsweringId: 'another-fake-test-id',
+        };
+
+        callListener();
```

Comment:
I think this test would successfully pass even with the previous implementation where we had the bug cause here in the `callListener` we are executing the listener that was added only by the first controller.

We need to make sure we are executing the listeners that were added by both controllers than confirm that the action was dispatched just once.

Reply:
héhé yes i was actually adding it right now